### PR TITLE
Fix time link hover

### PR DESCRIPTION
**Pull request by cydanil**
Gourmet, in its rich text widgets such as displaying the recipe, supports time links.
These time links are created by parsing the recipe and finding strings related to time, and create a link using Pango for rendering.
When clicked, these are meant to show a timer.

For a more intuitive usage, time links have the same behaviour as links in a browser: they are blue and underscored, and the cursor changes from an I-beam to a hande, as one would intuitively expect.

This fixes that, as shown below:
time_link

(When entering the TextView, the cursor becomes an I-beam, for text selection, and a hand when hovering a link).

Historically, one could set and check custom properties in a `Gtk.TextTag`. This is no longer allowed.
A workaround is to use `setattr` and `getattr` to set a flag:
```python
setattr(tag, 'is_a_link', True)
```

However, since tags will now be created automatically, setting custom attributes to a tag is difficult.

As Gourmet has a rather simple markup grammar, with only italic, bold, underlined, and blue text allowed, we can rely on the text color being blue to indicate a link.

### CODE REVIEW

File: gourmet/gtk_extras/LinkedTextView.py [hunk 1]
```diff
@@ -132,26 +133,32 @@ def event_after(self, text_view, event):
...
+                                  text_view: 'LinkedTextView',
+                                  x: int,
+                                  y: int) -> None:
+        """Set the mouse cursor to be a hand when hovering over a time link.
+
+        Check each tag covering the position (x, y) within the text view.
+        If the text happens to be blue, which only time links are allowed to be
+        in this application, then set the cursor to a hand.
+        If leaving the area of a time link, set the cursor back to the system's
+        default.
```
Comment (cydanil) [inline gourmet/gtk_extras/LinkedTextView.py]: If leaving the area of a time link, set the cursor to an I-beam

**Comment by maweki** [comment]
Does the color-stuff work if there's a different user theme where backgrounds and colors are swapped?

**Comment by cydanil** [comment]
Yes, the colouring is handled within the application:
image
(Screenshot with the Adwaita-dark theme)

Blue might not be the prettiest in such cases, but that's beyond this PR.
However, this can be made future proof, by checking that `foreground-gdk` is not None:
```python
if tag.get_property('foreground-gdk') is not None:
   ...
```

I maintain that this approach is okay, as only time links have coloured text.

**Comment by maweki** [comment]
> I maintain that this approach is okay, as only time links have coloured text.

I was not implying that it wasn't. I just didn't know whether we should check for blue or some user-defined style or how easily it breaks for dark-mode-style themes.

**Comment by cydanil** [comment]
Sure, it's a fair question :) And it rose a valid point: I think we should check that there's a colour, not that it's specifically blue.

**Comment by cydanil** [comment]
Regarding dark colour stuff, I came up with a solution that I propose in issue #139

File: gourmet/gtk_extras/LinkedTextView.py [hunk 2]
```diff
@@ -132,26 +133,30 @@ def event_after(self, text_view, event):
...
-        if hovering != self.hovering_over_link:
-            self.hovering_over_link = hovering
 
-        if self.hovering_over_link:
-            text_view.get_window(Gtk.TextWindowType.TEXT).set_cursor(self.hand_cursor)
-        else:
-            text_view.get_window(Gtk.TextWindowType.TEXT).set_cursor(self.regular_cursor)
+        cursor = self.text_cursor
+        if hovering_over_link:
+            cursor = self.hand_cursor
```
Comment (maweki) [inline gourmet/gtk_extras/LinkedTextView.py]: sometimes I feel `cursor = self.hand_cursor if hovering_over_link else self.text_cursor` is more concise as you don't have to mentally parse whether the `if` does anything intelligent or noteworthy. It is clear that it's just about the state of `cursor`.

But that's just taste and if you're happy with this PR, please merge it.

**Review by maweki — APPROVED** [review]

**Comment by cydanil** [comment]
I remember having a reason to prefer the multiline if, but I forgot what it was, and now I agree with you, so I went for it.

Thanks for taking the time to go through this!

**Pull request merged**